subt/ros/proxy/ros_proxy_node.cc

```diff
@@ -429,6 +429,10 @@ bool parseArtf(char *input_str, subt::msgs::Artifact& artifact)
         {
           type = subt::ArtifactType::TYPE_ROPE;
         }
+        if(strcmp(buf, "TYPE_CUBE") == 0)
```

Comment:
The time of rethinking the if-if-if structure was long ago.

If subt::ArtifactType was a protocol buffer enum, we would be subt::ArtifactType_Parse away from replacing the whole block. But it isn't. So ...

We can either use different strings and reuse [an existing conversion](https://github.com/osrf/subt/blob/53274189769f6c36497105ca6d864ac16b477c2c/subt_ign/include/subt_ign/Common.hh#L77), or roll our own:


```
// If we were willing to go all the way to C++17, I would consider returning
// std::optional<subt::ArtifactType> instead of returning a bool and having
// an output function parameter.
bool artifactTypeFromString(const std::string& type_text,
                            subt::ArtifactType& type_enum) {
  // This may need c++11 or higher. I hope we support that, because anything
  // below that is torturing ourselves without needing to.
  //
  // The static initialization also assumes single-threading. If multiple threads
  // can call the conversion function at the same time, we can remove `static`
  // and pay the cost of the function building the map at every call. Or do
  // something else about it.
  //
  // In big O notation, std::unordered_map is a better choice than std::map.
  // But this map is so small, that I am not sure if it is the case right here.
  static const std::map<std::string, subt::ArtifactType> conversion = {
    {"TYPE_BACKPACK", subt::ArtifactType::TYPE_BACKPACK},
    // ...
    {"TYPE_CUBE", subt::ArtifactType::TYPE_CUBE},
  };
  const auto type_lookup = conversion.find(type_text);
  if (type_lookup == conversion.end()) {
    return false;
  }
  type_enum = type_lookup.second;
  return true;
}

...

  subt::ArtifactType type;
  if (!artifactTypeFromString(buf, type)) {
    // report an error, return
  }
  artifact.set_type(static_cast<uint32_t>(type));
```